src/app/shared/directives/infiniteScroll/infinite-scroll/infinite-scroll.directive.ts

```diff
@@ -0,0 +1,21 @@
+import {Directive, EventEmitter, HostListener, Output} from '@angular/core';
+import ScrollDirection from './scroll-direction.enum';
+
+@Directive({
+    selector: '[appInfiniteScroll]',
+})
+export class InfiniteScrollDirective {
+    @Output() loadData = new EventEmitter<ScrollDirection>();
+    @HostListener('scroll', ['$event.target']) scroll(elem: HTMLElement) {
+        const {scrollHeight, scrollTop, offsetHeight} = elem;
```

Comment:
Я бы предложил перенести деструктуризацию в аргументы, довольно таки удобный подход
```suggestion
    @HostListener('scroll', ['$event.target']) scroll({
        scrollHeight,
        scrollTop,
        offsetHeight,
    }: HTMLElement) {
```